### ISSUE TITLE
Copy and Paste functions are broken

### ISSUE THREAD

**Issue by gravelfreeman** [CollaboraOnline/online#9281]
# Chomium and Firefox browsers

### Describe the Bug
It's not possible to copy content from within the Text, Calc and Prensentation apps to the clipboard.

### Steps to Reproduce
1. Copy some text in Windows clipboard
2. Paste text in Text, Calc and Prensentation

### Expected Behavior
Text gets copied in Text, Calc and Prensentation.

### Actual Behavior
Text doesn't get copied in Text, Calc and Prensentation.

***

# Chromium browsers only

### Describe the Bug
It's not possible to paste content from the clipboard into Text, Calc and Prensentation apps.

### Steps to Reproduce
1. Copy some text in Windows clipboard
2. Paste text in Text, Calc and Prensentation

### Expected Behavior
Text gets copied from Text, Calc and Prensentation to the clipboard.

### Actual Behavior
Nothing gets added to the clipboard.

***

### Desktop
```
COOLWSD version: 24.04.3.1 (git hash: 594b605 (E))
LOKit version: Collabora Office 24.04.3.1 (git hash: 31b6e2c)
Served by: Debian GNU/Linux 12 (bookworm)
```

### Additional Context
First shared on Nextcloud [forum ](https://help.nextcloud.com/t/copy-paste-issue-with-code-maybe-related-to-http-headers/195011)where I've been told to post my issue here.

**Comment by mulzhang** [comment]
I am also facing this tricky problem now, please solve it as soon as possible. Thank you

**Comment by HLFH** [comment]
I can reproduce the issue on Safari.

**Comment by timur-g** [comment]
There is no general Copy and Paste bug. If it is not working for you, you need to test more to figure out what exactly is not working: (simple text, complex text with "Download to clipboard" button), are there errors in browser console, are you behind some proxy... 
This report was initially Windows. The other user said "me too" wihtout specifics, so we may only assume that is the same. 
Safari may be different issue, we do not know now unless this is pinpointed.

**Comment by pedropintosilva** [comment]
Thank @gravelfreeman for reporting this and @mulzhang . Also thanks @HLFH for posting additional info in https://github.com/nextcloud/richdocuments/issues/3775#issuecomment-2191270588 WRT to version. 

Could any of you be so kind to screenrecord it (if possible, with browser console opened)? or Maybe easier, you could show up in the Weekly COOL Community Meeting and we could do it together during the testing part of the meeting. I happens Every Thursday 🕙 ~ 12:00 CEST at 💬 https://meet.jit.si/COOLWeeklyMeeting

**Comment by wbonis** [comment]
I also looked at this Bug:

I noticed the following:

- Behind traefik as reverse proxy 24.04.x does not allow copy and paste
- Behind traefik as reverse proxy 23.05.9.2.1 does work!
- If i use Caddy as reverse proxy, every version works!

But I can not see a difference in the requests / config between Caddy and Traefik

Maybe this helps to find the bug!

**Comment by apoeteo** [comment]
I'm also facing this issue, both Firefox- and Chromium-based browser, self hosted Nextcloud (AIO) behind Traefik. Will try to gather more detailed info. Thank you.

**Comment by P4z** [comment]
I have just experienced problems with copying text from .odt document after upgrading my NextCloud AIO containers to most recent version. I have not noticed any problem with copying in the past, but I was not using the Properties dialogue before.

I opened File => Properties to set a title of a document, then wanted to copy the very first paragraph that is formatted as Title style with mouse context menu, suddenly this popup raised:

![obraz](https://github.com/user-attachments/assets/efeb135b-a229-48de-a8a4-5ec4d049dad5)

When I use CTRL+C, nothing is copied and no popup shows.

When I only close the properties window then I can easily copy the text.

Not sure if this is related to this problem, but I think it is worth to mention my case. To me, this is of course a bug.

**Comment by timur-g** [comment]
Could someone please confirm this is still the issue with at least 24.04.7.1.2 (or 24.04.7.2.1)?

**Comment by timur-g** [comment]
There were some changes here. If someone builds, please test. Otherwise, changes will be in 24.04.8.1.

**Comment by mlan** [comment]
Traffic 3.1.5
Collaborate 24.04.8.1

The problem still persists.

**Comment by gokaysatir** [comment]
Hello @mlan thank you for re-checking the issue. The changes require you to reset the page-cookies. Can you try that if you haven't already.

We will also change that behaviour but for now, resetting the page-cookies should help. Thank you.

**Comment by mlan** [comment]
Hi @HLFH.

I deleted all cookies in the bowser and opened a document using Collabora and tried to copy from it. But the clipboard remained empty.

**Comment by jazevedo-coll** [comment]
Hi @mlan @apoeteo would it be possible for you to send us your traefik configurations, `config.yml` and `docker-compose` with PII and sensitive data removed. 

If needed we can also provide a way for you to send us the information outside if this ticket.

**Comment by jazevedo-coll** [comment]
A fix for the issue with traefik is included in the next COOL release (COOL 24.04.8.5), it was tested with traefik 3.1.5.
It also has improvements for the clipboard. 

These fixes will also be included in the next CODE (Collabora Online Development Edition) release. 

If the issue persists after updating, please reopen the issue. 

https://github.com/CollaboraOnline/online/pull/10413

https://github.com/CollaboraOnline/online/pull/10394

https://github.com/CollaboraOnline/online/pull/10190

**Comment by apoeteo** [comment]
@jazevedo-coll Sorry I didn't follow up earlier, I confirm the issue is fixed now, thanks a lot!!
